Issue building dnn modules

When building OpenCV contrib modules, especially dnn_modern, with latest versions of both master and contrib repo, I get the following error at 48% of build:

> /home/spu/Documents/github/opencv_contrib_dev/modules/dnn_modern/src/caffe_converter.cpp:185:33: error: no matching function for call to ‘tiny_dnn::network<tiny_dnn::sequential>::predict()’
>      auto result = net_->predict();
>                                  ^

Anyone has an idea on how to continue? For now I will disable the tiny_dnn inclusion.

Then when disabling, building breaks down at

```
Building CXX object modules/cudalegacy/CMakeFiles/opencv_test_cudalegacy.dir/test/TestDrawRects.cpp.o
/home/spu/Documents/github/opencv_dev/modules/stitching/src/cuda/multiband_blend.cu(93): error: Within a __device__/__global__ function, only __shared__ variables may be marked "static"

[ 90%] Built target opencv_test_cudaimgproc
1 error detected in the compilation of "/tmp/tmpxft_00004d79_00000000-7_multiband_blend.cpp1.ii".
```

From this[ Q&A](http://answers.opencv.org/question/128555/error-with-multiband_blendcu-compiling-opencv-320-on-ubuntu-1510-with-cuda-75/) it seems I am not the only one with this issue.

--- a/modules/dnn_modern/src/caffe_converter.cpp
+++ b/modules/dnn_modern/src/caffe_converter.cpp
@@ -182,7 +182,7 @@ void CaffeConverter_Impl::eval(const cv::InputArray image,
     const vector<tiny_dnn::float_t> vec(inputvec.begin(), inputvec.end());
 
     // perform inderence
-    auto result = net_->predict();
+    auto result = net_->predict(vec);
 
     // allocate output
     results->clear();

